esp-wifi/src/preempt/mod.rs

```diff
@@ -1,49 +1,102 @@
-macro_rules! sum {
-    ($h:expr) => ($h);
-    ($h:expr, $($t:expr),*) =>
-        ($h + sum!($($t),*));
-}
+#[cfg_attr(target_arch = "riscv32", path = "preempt_riscv.rs")]
+#[cfg_attr(target_arch = "xtensa", path = "preempt_xtensa.rs")]
+pub mod arch_specific;
+
+use arch_specific::*;
+use esp_wifi_sys::include::malloc;
+
+use crate::{compat::malloc::free, hal::trapframe::TrapFrame, memory_fence::memory_fence};
+
+static mut CTX_NOW: *mut Context = core::ptr::null_mut();
```

Comment:
Should we wrap these in a `Mutex<RefCell>` to actually ensure we always access via a CS? Would the perf penalty be too great?

If we don't, we should at least add a doc comment here stating these globals are shared across different threads of execution and should have guarded access.

Reply:
Could be [AtomicPtr](https://docs.rs/portable-atomic/1.7.0/portable_atomic/struct.AtomicPtr.html)

Reply:
Seems to not affect performance 👍